server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -293,6 +310,11 @@ public DesiredBalance compute(
                 for (final var shardRouting : routingNode) {
                     if (shardRouting.initializing()) {
                         hasChanges = true;
+                        if (shardRouting.primary()
+                            && shardRouting.unassignedInfo() != null
+                            && shardRouting.unassignedInfo().reason() == UnassignedInfo.Reason.INDEX_CREATED) {
```

Comment:
I wonder if we should consider another reasons?
Could it be an issue for grow/shrink or snapshot restore?
Regardless this could be a followup after a dedicated discussion.

Reply:
Yeap, exactly, the index creation happening during the ILM rolover